users.go

```diff
@@ -556,6 +556,55 @@ func (api *Client) SetUserRealNameContextWithUser(ctx context.Context, user, rea
 	return response.Err()
 }
 
+// SetUserCustomFields sets Custom Profile fields on the provided users account.  Due to the non-repeating elements
+// within the request, a map fields is required.  The key in the map signifies the field that will be updated.
+//
+// Note: You may need to change the way the custom field is populated within the Profile section of the Admin Console from
+// SCIM or User Entered to API.
+//
+// See GetTeamProfile for information to retrieve possible fields for your account.
+func (api *Client) SetUserCustomFields(userID string, customFields map[string]UserProfileCustomField) error {
+	return api.SetUserCustomFieldsContext(context.Background(), userID, customFields)
+}
+
+// SetUserCustomFieldsContext will set a users custom profile field with context.
+//
+// For more information see SetUserCustomFields
+func (api *Client) SetUserCustomFieldsContext(ctx context.Context, userID string, customFields map[string]UserProfileCustomField) error {
+
+	// Convert data to data type with custom marshall / unmarshall
+	// For more information, see UserProfileCustomFields definition.
+	updateFields := UserProfileCustomFields{}
+	updateFields.SetMap(customFields)
+
+	// This anonymous struct is needed to set the fields level of the request data.  The base struct for
+	// UserProfileCustomFields has an unexported variable named fields that does not contain a struct tag,
+	// which has resulted in this configuration.
+	profile, err := json.Marshal(&struct {
+		Fields UserProfileCustomFields `json:"fields"`
+	}{
+		Fields: updateFields,
+	})
```

Comment:
According to the following document, It seems that you don't need to wrap by "fields" field 🤔 

https://api.slack.com/methods/users.profile.set

Reply:
Under the section on that page titled "Custom profile fields", the follow example request is displayed:

```{
    "profile": {
        "fields": {
            "Xf0111111": {
                "value": "Barista",
                "alt": ""
            },
            "Xf0222222": {
                "value": "2022-04-11",
                "alt": ""
            },
            "Xf0333333": {
                "value": "https://example.com",
                "alt": ""
            }
        }
    }
}```

Which does have the fields section within Profile.  I've tested this code against my production slack workspace and it works as expected as well.

Reply:
Sorry, I was looking at another part of the document...